Automattic/wp-calypso
# Hosting metrics: abstract `useGroupByTime` to generate the data for the second line chart

- Related to https://github.com/Automattic/wp-calypso/pull/80382

## Proposed Changes
@katinthehatsite, this PR unifies our work grouping and counting the status codes in the server response.

* Abstract `useGroupByTime` to be used on the second line chart

## Testing Instructions



* Navigate to `/site-monitoring/${siteSlug}`
* Scroll down
* Observe the Bar Chart and the second chart display the data correctly.

## Pre-merge Checklist
